lightning/src/ln/channelmanager.rs

```diff
@@ -4748,6 +4801,23 @@ where
 					log_error!(self.logger, "Got a duplicate pending claimable event on payment hash {}! Please report this bug",
 						log_bytes!(payment_hash.0));
 				}
+
+				if let Some(RecipientOnionFields { ref custom_tlvs, .. }) = payment.onion_fields {
+					if !custom_tlvs_known && custom_tlvs.iter().any(|(typ, _)| typ % 2 == 0) {
+						log_info!(self.logger, "Cannot accept payment with unknown even TLVs. Rejecting payment with payment hash {}",
```

Comment:
The unknown even TLVs could be printed in the `log_info` by displaying them as a vec.

Reply:
Cool.